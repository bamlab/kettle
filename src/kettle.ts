export { kettleContent as content } from './kettleContent';
export { kettleFile as file } from './kettleFile';
export { kettlePath as path } from './kettlePath';
export { kettleTransformFactory as transformFactory } from './kettle.transform';
export { KettleOptions, KettleValues } from './kettle.types';
export { replaceIosUUIDs } from './iosUUID.replace';
export { iosUUIDsTransformFactory } from './iosUUID.transform';
export { gitignoreToGlob } from './gitignoreToGlob';
