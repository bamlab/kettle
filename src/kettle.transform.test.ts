import { src } from 'gulp';
import { kettleTransformFactory } from './kettle.transform';
import { resolve } from 'path';
import through2 = require('through2');
import * as Vinyl from 'vinyl';

interface TestContent {
  [key: string]: 'file' | 'folder';
}

describe('kettleTransformFactory', () => {
  it('works against a template', jestCallback => {
    const inputPath = resolve(__dirname, '../testFiles/templateInput');
    const content: TestContent = {};

    const contentCollector = through2.obj(
      function(chunk: Vinyl, encoding, callback) {
        const chunkPath = chunk.path.replace(inputPath, '');
        if (chunk.isNull()) content[chunkPath] = 'folder';
        if (chunk.isBuffer()) {
          content[chunkPath] = 'file';
          expect(chunk.contents.toString()).toMatchSnapshot(chunkPath);
        }
        callback();
      },
      function() {
        expect(content).toMatchSnapshot('File architecture');
        jestCallback();
      }
    );

    src([`${inputPath}/**/*`])
      .pipe(
        kettleTransformFactory({
          values: {
            isDebug: false,
            isProd: true,
            isTrue: true,
            isFalse: false,
            utilsName: 'myUtils',
            appName: 'myApp',
          },
        })
      )
      .pipe(contentCollector);
  });
});
