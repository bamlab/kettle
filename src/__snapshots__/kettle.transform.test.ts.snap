// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kettleTransformFactory works against a template: /myApp.js 1`] = `
"const test = require('./shouldExist/shouldExist.js');
"
`;

exports[`kettleTransformFactory works against a template: /package.json 1`] = `
"{
  \\"name\\": \\"myApp\\"
}
"
`;

exports[`kettleTransformFactory works against a template: /shouldExist/shouldExist.js 1`] = `
"module.exports = () => {};
"
`;

exports[`kettleTransformFactory works against a template: /src/myUtils/index.js 1`] = `
"import hello from 'world';

function myUtils(random) {
  setupProd(random);
  return hello();
}
"
`;

exports[`kettleTransformFactory works against a template: /src/myUtilsmyApp.js 1`] = `""`;

exports[`kettleTransformFactory works against a template: File architecture 1`] = `
Object {
  "/myApp.js": "file",
  "/package.json": "file",
  "/shouldExist": "folder",
  "/shouldExist/shouldExist.js": "file",
  "/src": "folder",
  "/src/myUtils": "folder",
  "/src/myUtils/index.js": "file",
  "/src/myUtilsmyApp.js": "file",
}
`;
